ngx_http_upload_module.c

```diff
@@ -2514,7 +2514,11 @@ ngx_http_upload_merge_ranges(ngx_http_upload_ctx_t *u, ngx_http_upload_range_t *
     }
 
     if(out_buf.file_pos < state_file->info.st_size) {
-        (void) ftruncate(state_file->fd, out_buf.file_pos);
+        int res = ftruncate(state_file->fd, out_buf.file_pos);
+        if(res == EBADF || res == EINVAL) {
+            rc = NGX_ERROR;
+            go failed;
```

Comment:
@hongzhidao `go` MUST be `goto`, or it CANNOT be compiled successfully.

Reply:
@winshining Fixed, thanks for reporting.